Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp

```diff
@@ -721,6 +727,113 @@ GstClockTime webkitGstElementGetCurrentRunningTime(GstElement* element)
 }
 #endif
 
+static bool checkNeedsAppsinkWorkaround()
+{
+    GRefPtr<GstElementFactory> factory = adoptGRef(gst_element_factory_find("appsink"));
+    if (!factory) {
+        WTFLogAlways("GStreamer element appsink not found. Please install it");
+        return false;
+    }
+
+    GRefPtr<GstPlugin> plugin = adoptGRef(gst_plugin_feature_get_plugin(GST_PLUGIN_FEATURE(factory.get())));
+    if (!plugin) {
+        WTFLogAlways("appsink not part of a plugin. Registered appsink workaround twice?");
+        ASSERT_NOT_REACHED("appsink not part of a plugin");
+        return false;
+    }
+    const char* versionString = gst_plugin_get_version(plugin.get());
```

Comment:
I wonder if it'd be better to use `gst_plugins_base_version()` (appsink is part of base), because it won't be necessary to parse the output string.

Reply:
Oh, I wasn't aware of that function. Neat.